src/modules/PublicLab.CustomInsert.js

```diff
@@ -0,0 +1,100 @@
+ module.exports = function CustomInsert(_module, wysiwyg) {
+	function Syntax(tag ,Option1, Option2) {
+		if(Option2 == "List") {
+			if(Option1 == "Notes") {
+				var syn = "[notes:" + tag + "]";
+			}
+			if(Option1 == "Wikis") {
+				var syn = "[wikis:" + tag + "]";
+			}
+			if(Option1 == "Nodes(Wikis + Notes)") {
+				var syn = "[nodes:" + tag + "]";
+			}
+			if(Option1 == "Activity") {
+				var syn = "[activity:" + tag + "]";
+			}
+			if(Option1 == "Questions") {
+				var syn = "[questions" + tag + "]";
+			}
+		}
+		if(Option2 == "Grid") {
+			if(Option1 == "Notes") {
+				var syn = "[notes:grid" + tag + "]";
+			}
+			if(Option1 == "Wikis") {
+				var syn = "[wikis:grid" + tag + "]";
+			}
+			if(Option1 == "Nodes(Wikis + Notes)") {
+				var syn = "[nodes:grid" + tag + "]";
+			}
+			if(Option1 == "Activity") {
+				var syn = "[activity:grid" + tag + "]";
+			}
+			if(Option1 == "Questions") {
+				var syn = "[questions:grid" + tag + "]";
+			}
+		}
+		return syn;
+	}
+
+   $('.wk-commands').append('<a class="woofmark-command-insert btn btn-default" data-toggle="Insert" title="Custom Insert"><i class="fa fa-tags"></i></a>');
+
+   var builder = '<div class="dropdown" style="margin-bottom: 20px;">';
+   builder += '<button class="btn btn-default dropdown-toggle" type="button" id="dropdownMenu1" data-toggle="dropdown" aria-haspopup="true" aria-expanded="true" style="min-width: 150px;" >'
+   builder += '<span id= "selected">What Do you want to insert?</span>';
+   builder += '<span class="caret"></span>';
+   builder += '</button>';
+   builder += '<ul class="dropdown-menu" role="menu" aria-labelledby="dropdownMenu1" id="menu">';
+   builder += '<li role="presentation"><a role="menuitem" tabindex="-1" id="Notes">Notes</a></li>';
```

Comment:
Hi, I wonder if it may be better to use a classname so as not to have to use a more unique `id`, and to nest the selectors, so you'd be able to call: `. woofmark-command-insert .notes` instead of relying on a completely unique `Notes` id? And similarly for other `id`s here! What do you think?

Reply:
Ok. Making changes

Reply:
Are you saying to change id into class. How would that be beneficial? :thinking:

Reply:
well, if you use an id, then if there are 2 editors on the page, they're no longer unique, and thats a rule with ids. It may be better to use nested classes for specificity. This may not come up much, but in those edge cases it's a good practice. And, the `id` "menu" is not very unique so it already may collide with `id`s in other code. So i guess I'm just noting that we could bypass the uniqueness thing entirely by relying more on classes. Thanks!